src/components/Receiving/ItemDetails.js

```diff
@@ -179,11 +182,16 @@ class ItemDetails extends Component {
 
       selectedItem[key] = value;
 
+      if (key === 'barcode') {
+        if (!item.isSelected && value.length > 0) this.toggleItem(item);
+        if (item.isSelected && value.length === 0) this.toggleItem(item);
```

Comment:
I think there might be situations when value could came as `null` or `undefined`. Don't you think so?

Reply:
also I've checked the ticket and found no requirement to clear selection

Reply:
I don't think so. We don't get these values ​​from elements with the values and if we don't sent manually, but for security, I will add one more condition